walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,102 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+
+  /**
+   * Computes a list of all time ranges where all attendees of a meeting are busy
+   * @param  events    Collection of Event objects formed before the creation of this meeting
+   * @param  attendees Collection of attendee string names
+   * @return List of time ranges where attendees are engaged in other events.
+   */
+  private List<TimeRange> getBusyTimeRanges(Collection<Event> events, Collection<String> attendees) {
+    List<TimeRange> busyTimeRanges = new ArrayList<TimeRange>();
+    for (Event event : events) {
+      Set<String> currEventAttendees = event.getAttendees();
+      for (String attendee : attendees) {
+        if (currEventAttendees.contains(attendee)) {
+          busyTimeRanges.add(event.getWhen());
+          break;
+        }
+      }
+    }
+    return busyTimeRanges;
+  }
+
```

Comment:
In general, put the public methods before private methods. Private methods are usually helper methods, and they are difficult to understand without the context

Reply:
Done.